libs/ledger/src/upow/naive_synergetic_miner.cpp

```diff
@@ -165,6 +168,23 @@ WorkPtr NaiveSynergeticMiner::MineSolution(Digest const &        contract_digest
                                            chain::Address const &contract_address,
                                            ProblemData const &   problem_data)
 {
+  // TODO(AB): State sharding needs to be added here
+  BitVector shards{num_lanes_};
+  shards.SetAllOne();
+  StateSentinelAdapter storage_adapter{storage_, Identifier{"fetch.token"}, shards};
```

Comment:
This should be just a state adapter. This will ensure that any operations from this will be read only